aea/aea_builder.py

```diff
@@ -1201,7 +1204,7 @@ def set_from_configuration(
         self.set_loop_mode(agent_configuration.loop_mode)
         self.set_runtime_mode(agent_configuration.runtime_mode)
 
-        if agent_configuration._default_connection is None:
+        if agent_configuration.default_connection is None:
```

Comment:
Note; the property only returns string; it cannot return None, so this is not valid; please revert